src/components/post/battle/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { useQuery } from '@tanstack/react-query';
+import { useRouter } from 'next/router';
+import { getPostBattleData } from './api';
+import MusicInfo from './musicinfo';
+import { PostBattleInfo } from './types';
+
+function PostBattle() {
+  const router = useRouter();
+  const { postId } = router.query;
+
+  const { data: battleMusic, isLoading } = useQuery<PostBattleInfo>(
+    ['post', 'battle', postId],
+    () => getPostBattleData(parseInt(postId as string)),
+    {
+      enabled: !!postId,
+    },
+  );
+
+  const navigatePostMyBattleList = () => {
+    const { music } = battleMusic as PostBattleInfo;
+    router.push(`/post/battle/mybattlelist?postId=${postId}&genre=${music.genre?.genreName}`);
+  };
+
+  return (
+    <>
+      {isLoading || !battleMusic ? (
+        <></>
+      ) : (
+        <>
+          <MusicInfo
+            musicName={battleMusic.music.musicName}
+            musicUrl={battleMusic.music.musicUrl}
+            thumbnailUrl={battleMusic.music.thumbnailUrl}
+            singer={battleMusic.music.singer}
+            genre={battleMusic.music.genre}
+          />
+          <button onClick={navigatePostMyBattleList}>내 대결 곡 고르기</button>
+        </>
+      )}
+    </>
```

Comment:
p3; 지금은 로딩 디자인이 따로 없으니까 그저 battleMusic이 있을 때에만 렌더링 하는 게 목적이잖아요 그럼 이렇게 하면 좋을 것 같아요

```suggestion
    <>
      { battleMusic && (
        <>
          <MusicInfo
            musicName={battleMusic.music.musicName}
            musicUrl={battleMusic.music.musicUrl}
            thumbnailUrl={battleMusic.music.thumbnailUrl}
            singer={battleMusic.music.singer}
            genre={battleMusic.music.genre}
          />
          <button onClick={navigatePostMyBattleList}>내 대결 곡 고르기</button>
        </>
      )}
    </>
```